compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/RootNodeOwner.skiko.kt

```diff
@@ -499,6 +500,31 @@ internal class RootNodeOwner(
     }
 }
 
+// the max value that can be passed as Constraints(0, LargeDimension, 0, LargeDimension)
+// greater values cause "Can't represent a width of"
+internal const val LargeDimension = (1 shl 14) - 1 // 16383
+
+/**
+ * After https://android-review.googlesource.com/c/platform/frameworks/support/+/2901556
+ * Compose core doesn't allow measuring in infinity constraints,
+ * but RootNodeOwner and ComposeScene allow passing Infinity constraints by contract
+ * (Android on the other hand doesn't have public API for that and don't have such an issue).
+ *
+ * This method adds additional check on Infinity constraints,
+ * and pass constraint large enough instead
+ */
+private fun MeasureAndLayoutDelegate.updateRootConstraintsWithInfinityCheck(
+    constraints: Constraints
+) {
+    val maxWidth = if (constraints.hasBoundedWidth) constraints.maxWidth else LargeDimension
```

Comment:
Doesn't Google also check for values larger than 2^24?

https://android-review.googlesource.com/c/platform/frameworks/support/+/2901556/5/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/node/LookaheadDelegate.kt#328

Reply:
We have a more strict constraint in the `Constraints` constructor. I changed it in the last commit to `Constraints.MinNonFocusMask - 1` and added an explanation.